napi_memory_management.md

```diff
@@ -0,0 +1,143 @@
+# Memory Management
+
+As N-API calls are made, references to objects in the heap for the underlying
+vm may be returned as ```napi_value```.  These references must hold the
+objects 'live' until they are no longer required by the native code,
+otherwise the objects could be collected before the native code was
+finished using them.
+
+As object references are returned they are associated with a
```

Comment:
Should this doc use the term "handles" instead of "references"? Unless specifically referring to persistent or weak (`napi_ref`) references.